api/Dockerfile

```diff
@@ -0,0 +1,32 @@
+# FROM golang:latest AS builder
+FROM registry.access.redhat.com/ubi9/ubi-minimal:latest AS builder
+
+ARG TARGETARCH
+USER root
+RUN microdnf install -y tar gzip unzip
+
+# install platform specific go version
+RUN curl -O -J  https://dl.google.com/go/go1.22.0.linux-${TARGETARCH}.tar.gz
+RUN tar -C /usr/local -xzf go1.22.0.linux-${TARGETARCH}.tar.gz
+RUN ln -s /usr/local/go/bin/go /usr/local/bin/go
+
+
+# Install protoc
+RUN curl -L -o protoc.zip https://github.com/protocolbuffers/protobuf/releases/download/v27.1/protoc-27.1-linux-x86_64.zip \
+    && unzip protoc.zip -d /usr/local/ && rm protoc.zip
+
+# Install protoc plugins
+RUN go install google.golang.org/protobuf/cmd/protoc-gen-go@latest
+RUN go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@latest
+RUN go install github.com/go-kratos/kratos/cmd/protoc-gen-go-http/v2@latest
+RUN go install github.com/google/gnostic/cmd/protoc-gen-openapi@latest 
+RUN go install github.com/envoyproxy/protoc-gen-validate@latest
+
+FROM registry.access.redhat.com/ubi9/ubi-minimal:latest
```

Comment:
Need to use the same UBI image as other files 8.X (IIRC 8.10 refer: https://github.com/project-kessel/relations-api/blob/main/Dockerfile#L1)